src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -108,6 +110,18 @@ bool RaftReplDev::join_group() {
 
 void RaftReplDev::use_config(json_superblk raft_config_sb) { m_raft_config_sb = std::move(raft_config_sb); }
 
+void RaftReplDev::on_create_snapshot(nuraft::snapshot& s, nuraft::async_result< bool >::handler_type& when_done) {
+    RD_LOG(DEBUG, "create_snapshot last_idx={}/term={}", s.get_last_log_idx(), s.get_last_log_term());
+    repl_snapshot snapshot{.last_log_idx_ = s.get_last_log_idx(), .last_log_term_ = s.get_last_log_term()};
+    auto result = m_listener->on_create_snapshot(snapshot).get();
```

Comment:
I think we need to ask upper layer to create the snapshot and require upper layer snapshot to be derived from repl_snapshot and store required parameters like last_term and last_log_idx. This way how snapshot created and maintained is completely controlled by upper layer.